hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1809,6 +1809,10 @@ public boolean balance(boolean force) throws IOException {
 
       List<RegionPlan> plans = this.balancer.balanceCluster(assignments);
 
+      if (runMode.isDryRun()) {
+        return true;
```

Comment:
Maybe outside of the scope of this work, but it would be great, IMO, to pass back details on what the balancer actually did through the `balancer` RPC. Simple output could just be "moved N regions". A verbose option could actually enumerate the plan changes. That'd be way nicer than having to hunt-peck in the master log after you make this call.

Reply:
Yea, that's a great idea. We were discussing something similar in the jira, though I'd prefer to do that in a separate jira/PR since it'd be another involved change.

Reply:
Heh, yeah, I saw the chatter on Jira after I posted the review 🤐